velociraptor/autoplotter/lines.py

```diff
@@ -290,8 +291,186 @@ def create_line(
 
         return self.output
 
+    def highlight_data_outside_domain(
+        self,
+        ax: Axes,
+        x: unyt_array,
+        y: unyt_array,
+        color: str,
+        x_lim: List,
+        y_lim: List,
+    ) -> None:
+
+        """
+        Add arrows to the plot for each data point residing outside the plot's domain.
+        The arrows indicate where the missing points are. For a given missing data point
+        with its Y(X) coordinate outside the Y(X)-axis range, the corresponding arrow
+        will have the same X(Y) coordinate and point to the direction where the missing
+        point is. If a data point happens to lie outside both the X-axis range and
+        Y-axis range, then a diagonal arrow is drawn.
+
+        Parameters
+        ----------
+
+        ax: Axes
+            An object of axes where to draw the arrows
+
+        x: unyt_array
+            Horizontal axis data
+
+        y: unyt_array
+            Vertical axis data
+
+        color: str
+            Color of the arrows that this function will draw. The color should be the
+            same as the color of the (missing) data points.
+
+        x_lim: List
+            A 2-length list containing the lower and upper limits of the X-axis range.
+
+        y_lim: List
+            A 2-length list containing the lower and upper limits of the Y-axis range.
+        """
+
+        # Additional check to ensure all provided data points are good
+        if not isnan(x).any() and not isnan(y).any():
+
+            # Arrow parameters
+            arrow_length = 0.07
+            distance_from_edge = 0.01
+
+            # Split data into three categories (along X axis)
+            below_x_range = x < x_lim[0]
+            above_x_range = x > x_lim[1]
+            within_x_range = (x > x_lim[0]) * (x < x_lim[1])
```

Comment:
I'm not sure about this style... Can we do `logical_and(x < xlim[0], x > x_lim[1])`

Reply:
Sure!

Should be in the code now